pkg/chrootarchive/archive_unix.go

```diff
@@ -79,6 +80,11 @@ func invokeUnpack(decompressedArchive io.Reader, dest string, options *archive.T
 	w.Close()
 
 	if err := cmd.Wait(); err != nil {
+		// when `xz -d -c -q | docker-untar ...` failed on docker-untar side,
+		// we need to exhaust `xz`'s output, otherwise the `xz` side will be
+		// pending on write pipe forever
+		io.Copy(ioutil.Discard, decompressedArchive)
```

Comment:
I'm trying to repro with this: https://gist.github.com/cpuguy83/eeb006f9cbc3b80567d7

Where I'm calling `xz -d -c -q foo.tar.xz | crash.go`